Manage bookings UI 19

iamalx opened the pull request:
### What this PR does (required):
- UI for sitters page: list of request , calendar, and button to edit

### Screenshots / Videos (required):
- ![sitters](https://user-images.githubusercontent.com/39109614/133964747-76c3ad4e-22da-4e2e-8505-4e6e74d5bf51.PNG)
### Any information needed to test this feature (required):
- need to install calendar package(WIP) 

### Any issues with the current functionality (optional):
- Calendar module does not have TS file which is causing issue - TODO.
- need to set up a new module with the adjustments from the original calendar package
- Some files were commit that need to be uncommitted  - need to delete?
- Need to set up images to set the imgs/avatar

iamalx committed:
made calendar work => finsih ui

iamalx committed:
main ui done => edits

iamalx committed:
finalized ui => add status

iamalx committed:
prep to add data from api

iamalx committed:
need popover

iamalx committed:
added button to edit

iamalx committed:
pr edits

iamalx committed:
pr edits 2

ajaytomgeorge commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,66 @@
+
+export default function Bookings(): JSX.Element {
+  const classes = useStyles();
+  // temp {} to hold data from API request.
```
Hey, you can remove this comment,

iamalx replied:
hi @ajaytomgeorge. yes I can remove it, but the reason it's there is to leave a note when this UI gets integrated with the BE. I will take this recommendation for future commits .

ajaytomgeorge commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,66 @@
+export default function Bookings(): JSX.Element {
+  const classes = useStyles();
+  // temp {} to hold data from API request.
+  const requests: any = {
```
You can skip the usage of _any_ as its not a recommended best practice

iamalx replied:
okay I will added interfaces

iamalx replied:
changed in Integration issue

ajaytomgeorge commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,66 @@
+
+  return (
+    <React.Fragment>
+      <CssBaseline />
```
CssBaseline is a global reset so including the CssBaseline in the App is the right way to go :-)

iamalx replied:
okay, changed

ajaytomgeorge commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,66 @@
+    <React.Fragment>
+      <CssBaseline />
+      <Container maxWidth="md" className={classes.container}>
+        <div className={`${classes.root} ${classes.mt}`}>
```
Is there any reason in why you didn't rely on _Box_ component

iamalx replied:
I changed all divs to Box

ajaytomgeorge commented on client/src/pages/Bookings/BookingsTemplate.tsx:
```diff
@@ -0,0 +1,136 @@
+}
+
+export default function ManageBookings({ bookings, sectionName }: Props): JSX.Element {
+  const pageStyles = makeStyles((theme: Theme) =>
```
Please move the css to another file

iamalx replied:
changed to css files for all

ajaytomgeorge commented on client/src/pages/Bookings/BookingsTemplate.tsx:
```diff
@@ -0,0 +1,136 @@
+                    </Typography>
+                    <Typography variant="body2" className={pageClasses.nameWrapper}>
+                      {/* TEMP IMG tags */}
+                      {/* <Avatar alt="Remy Sharp" src="../../Images/68f55f7799df6c8078a874cfe0a61a5e6e9e1687.png" /> */}
```
Please remove these commented out lines

ajaytomgeorge commented on client/src/pages/Bookings/useStyles.ts:
```diff
@@ -0,0 +1,24 @@
+// import { makeStyles } from '@material-ui/core/styles';
```
You can remove these commented line :-)

ajaytomgeorge commented on client/src/pages/Bookings/Calendar.tsx:
```diff
@@ -0,0 +1,20 @@
+import 'date-fns';
```
Is this import being used ?

iamalx replied:
yes as part of the calendar import - MultipleDatesPicker

ajaytomgeorge commented on client/src/pages/Bookings/EditButton.tsx:
```diff
@@ -0,0 +1,59 @@
+import Button from '@material-ui/core/Button';
+import SettingsIcon from '@material-ui/icons/Settings';
+
+const useStyles = makeStyles((theme: Theme) =>
```
You can shift CSS to another file

ajaytomgeorge commented on client/src/pages/Bookings/ManageBookings.tsx:
```diff
@@ -0,0 +1,33 @@
+
+export default function ManageBookings({ pastBookings, currentBookings }: Props): JSX.Element {
+  const pageStyles = makeStyles((theme: Theme) =>
+    createStyles({
```
CSS can shifted here too

ajaytomgeorge commented on client/src/pages/Bookings/NextBookings.tsx:
```diff
@@ -0,0 +1,101 @@
+import EditButton from './EditButton';
+
+interface Props {
+  nextBooking: any;
```
Any:-)

iamalx replied:
changed in integration issue

ajaytomgeorge commented on client/src/pages/Bookings/NextBookings.tsx:
```diff
@@ -0,0 +1,101 @@
+export default function NextBookings({ nextBooking }: Props): JSX.Element {
+  const pageStyles = makeStyles((theme: Theme) =>
+    createStyles({
+      paper: {
```
move this CSS too:-)

ajaytomgeorge reviewed (commented):
Good work ... Some minor comments though

rajivtitus commented on client/src/App.tsx:
```diff
@@ -23,6 +24,7 @@ function App(): JSX.Element {
                 <Route exact path="/dashboard">
                   <Dashboard />
                 </Route>
+                <Route exact path="/mysitter" component={Bookings} />
```
Please use hyphenated case for route names.

Eg: `edit-profile`

iamalx replied:
done

rajivtitus commented on client/src/pages/Bookings/BookingsTemplate.tsx:
```diff
@@ -0,0 +1,136 @@
+                    <Typography variant="body2" className={pageClasses.nameWrapper}>
+                      {/* TEMP IMG tags */}
+                      {/* <Avatar alt="Remy Sharp" src="../../Images/68f55f7799df6c8078a874cfe0a61a5e6e9e1687.png" /> */}
+                      <div className={pageClasses.tempImg}></div>
```
You can use MUI `Box` components instead of divs

Ditto throughout!

iamalx replied:
done

rajivtitus commented on client/src/pages/Bookings/ManageBookings.tsx:
```diff
@@ -0,0 +1,33 @@
+import Paper from '@material-ui/core/Paper';
+import BookingsTemplate from './BookingsTemplate';
+interface Props {
+  pastBookings: any;
```
Maybe use the booking interface here instead of type any?

rajivtitus commented on client/src/pages/Bookings/NextBookings.tsx:
```diff
@@ -0,0 +1,101 @@
+            <Grid item xs container direction="column" spacing={2}>
+              <Grid item xs>
+                <Typography variant="body2" gutterBottom className={pageClasses.sectionHr}>
+                  YOUR NEXT BOOKING
```
Please avoid using CAPITACASE as some screen readers interpret them as abbreviations. You can use a CSS class to transform the text instead.

iamalx replied:
okay, done.

rajivtitus replied:
Please address my comment from above and make the relevant change

rajivtitus commented on client/src/pages/Bookings/NextBookings.tsx:
```diff
@@ -0,0 +1,101 @@
+import EditButton from './EditButton';
+
+interface Props {
+  nextBooking: any;
```
Maybe use the booking interface here instead of type any?

Ditto throughout where applicable!

iamalx replied:
changed in Integrations issue

iamalx commented:
@rajivtitus  I have addressed all comments..
1) inline comments are there when I integrate to the BE. I have deleted some
2) ANY type will be addressed in Integrations Issue

iamalx committed:
pr addresed comments

rajivtitus commented on client/src/pages/Bookings/BookingStyles/BookingsTemplate.ts:
```diff
@@ -0,0 +1,74 @@
+      display: 'flex',
+      alignItems: 'center',
+      fontSize: '8px',
+      color: 'rgb(209,209,209)',
```
It is advisable to set colors in the theme and use the theme to apply them

Ditto throughout!

rajivtitus commented on client/src/pages/Dashboard/Dashboard.tsx:
```diff
@@ -5,8 +5,9 @@ import useStyles from './useStyles';
 import { useSocket } from '../../context/useSocketContext';
 import { useHistory } from 'react-router-dom';
-import ChatSideBanner from '../../components/ChatSideBanner/ChatSideBanner';
+// import ChatSideBanner from '../../components/ChatSideBanner/ChatSideBanner';
```
Please remove commented code throughout

iamalx replied:
done

rajivtitus reviewed (approved):
Just a couple of minor comments to address before merging.

As for the interfaces, you can apply them to the integration PR. Thanks!

iamalx committed:
final commit/pr comments/theme

iamalx commented:
@rajivtitus I added the changes. 

I accidentally added the package and yarn files and now have conflicts. what is the best way to address?

rajivtitus commented:
> @rajivtitus I added the changes.
> 
> I accidentally added the package and yarn files and now have conflicts. what is the best way to address?

You can follow along the article to resolve packgage.json conflicts: https://tkdodo.eu/blog/solving-conflicts-in-package-lock-json

Alternatively, you can also revert the commit: https://www.gitkraken.com/learn/git/problems/revert-git-commit
